executor/src/witgen/machines/machine_extractor.rs

```diff
@@ -66,7 +68,9 @@ pub fn split_out_machines<'a, T: FieldElement>(
         // Extract all witness columns in the RHS of the lookup.
         let lookup_witnesses = match id {
             Identity::Lookup(LookupIdentity { right, .. })
-            | Identity::Permutation(PermutationIdentity { right, .. }) => {
+            | Identity::PhantomLookup(PhantomLookupIdentity { right, .. })
+            | Identity::Permutation(PermutationIdentity { right, .. })
```

Comment:
@georgwiese I'm unsure whether these should include permutations

Reply:
Yes, they should! In general, I think witgen should largely treat phantom lookups / permutations like their native variant. In this case, it is looking at them to segment columns into machines.